detection-rules/impersonation_docusign.yml

```diff
@@ -65,27 +69,57 @@ source: |
   // identifies the main CTA in the email, eg "Review now" or "Review document"
   // this should always be a known docusign domain,
   // even with branded docusign subdomains
-  and any(body.links,
-          // we've observed invisible characters in the display name
-          // such as U+034f(look carefully): "Revi͏ew Now"
-          (
-            strings.ilevenshtein(.display_text, "Review Now") <= 3
-            or (
-              strings.icontains(.display_text, "Review")
-              and not strings.icontains(.display_text, "Review Us")
-            )
-            or strings.icontains(.display_text, "Now")
-            or strings.icontains(.display_text, "document")
+  and (
+    any(body.links,
+        // we've observed invisible characters in the display name
+        // such as U+034f(look carefully): "Revi͏ew Now"
+        (
+          strings.ilevenshtein(.display_text, "Review Now") <= 3
+          or (
+            strings.icontains(.display_text, "Review")
+            and not strings.icontains(.display_text, "Review Us")
           )
+          or strings.icontains(.display_text, "Now")
+          or strings.icontains(.display_text, "document")
+        )
         and not .href_url.domain.root_domain in ("docusign.com", "docusign.net")
         and not (
           .href_url.domain.root_domain == "mimecastprotect.com"
           and (
             .href_url.query_params is not null
-            and (regex.icontains(.href_url.query_params, 'domain=(?:\w+.)?docusign.net')
-            or regex.icontains(.href_url.query_params, 'domain=(?:\w+.)?docusign.com'))
+            and (
+              regex.icontains(.href_url.query_params,
+                              'domain=(?:\w+.)?docusign.net'
+              )
+              or regex.icontains(.href_url.query_params,
+                                 'domain=(?:\w+.)?docusign.com'
+              )
+            )
           )
         )
+    )
+    // Suspicious attachment
+    or any(attachments,
+           (
+             .file_extension in~ ("html", "htm", "shtml", "dhtml")
+             or .file_extension in~ $file_extensions_common_archives
+             or .file_type == "html"
+             or .content_type == "text/html"
+           )
+           and 1 of (
```

Comment:
I _think_ changing this to a series of `or` statements instead of a `1 of` allows it to bail out faster when one of them matches.

Reply:
I tested this theory recently and it didn't seem to matter.